test/test_nested_tensor_masking.py

```diff
@@ -39,312 +45,995 @@ def test_gen_nested_tensor(self):
+                               lambda: nt.nested_tensor_from_tensor_mask(tensor, mask))
+
+    def test_ntftm_single_scalar_mask_false(self):
+        scalar = torch.tensor([1], dtype=torch.uint8)
+        mask = torch.tensor(False)
+        self.assertRaisesRegex(RuntimeError, "Scalar mask cant be False.", lambda: nt.nested_tensor_from_tensor_mask(scalar, mask))
+
+    def test_ntftm_single_scalar_error(self):
+        tensor = torch.tensor(1)
+        mask = torch.tensor(True)
+        self.assertRaisesRegex(RuntimeError, "Can't construct nested tensor from a scalar.", lambda: nt.nested_tensor_from_tensor_mask(tensor, mask))
 
+    def test_ntftm_single_scalar(self):
+        tensor = torch.tensor([1])
+        mask = torch.tensor(True)
         res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask)
-        self.assertEqual(expected_nt0, res_nt)
+        self.assertEqual(res_nt, nt.nested_tensor([torch.tensor(1)]))
+        check_tensor_options(self, tensor, res_nt)
 
-        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask, nested_dim=0)
-        self.assertEqual(expected_nt0, res_nt)
+        mask = torch.tensor([True])
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask)
+        self.assertEqual(res_nt, nt.nested_tensor([torch.tensor(1)]))
+        check_tensor_options(self, tensor, res_nt)
 
-        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask, nested_dim=1)
-        self.assertEqual(expected_nt1, res_nt)
+        # Extra dim
+        tensor = torch.tensor([[1]])
+        mask = torch.tensor(True)
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask)
+        self.assertEqual(res_nt, 
+                        nt.nested_tensor([
+                            torch.tensor([1])
+                        ]))
+        check_tensor_options(self, tensor, res_nt)
 
-        # nested_dim is bigger than masks dimension 
-        self.assertRaises(ValueError, lambda: nt.nested_tensor_from_tensor_mask(tensor, mask, nested_dim=2))
-        self.assertRaises(ValueError, lambda: nt.nested_tensor_from_tensor_mask(tensor, mask, nested_dim=3))
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask, nested_dim=2)
+        self.assertEqual(res_nt, 
+                        nt.nested_tensor([
+                            nt.nested_tensor([
+                                torch.tensor(1)
+                            ])
+                        ]))
+        check_tensor_options(self, tensor, res_nt)
 
-    def test_nested_tensor_from_tensor_mask_dim_2(self):
-        original_nt = nt.nested_tensor([
-            nt.nested_tensor([
-                nt.nested_tensor([
-                    torch.tensor([[1, 2], 
-                                [3, 4]])
-                ]),
-            ]),
+    def test_ntftm_multi_scalars(self):
+        tensor = torch.tensor([1, 2, 3])
+        mask = torch.tensor(True)
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask)
```

Comment:
So in this case nested_dim is inferred from tensor.dim()?

Reply:
Or it's always 1 by default?

Reply:
its 1 by default